File: trees/ltcd_1376.cpp
//LeetCode - problem 1376

	int numOfMinutes(int n, int headID, vector<int>& manager, vector<int>& time) {
		vector<int> adj[n];
		for(int i=0;i<n;i++){
			if(i!=headID)adj[manager[i]].push_back(i);
		}
		queue<pair<int,int>> q;
		int ans = 0;
		q.push({headID,0});
		while(!q.empty()){
			int cur = q.front().first;
			int t = q.front().second;q.pop();

			for(auto& nxt:adj[cur]){
				ans = max(ans,t+time[cur]);
				q.push({nxt,t+time[cur]});
			}
		}
		return ans;
	}

File: trees/ltcd_1376.py
'''
LeetCode - problem 1376
'''

def numOfMinutes(self, n: int, headID: int, manager: List[int], informTime: List[int]) -> int:
        graph = collections.defaultdict(list)
        head = manager.index(-1)
        for emp,man in enumerate(manager):
            graph[man].append(emp)
        q = collections.deque()
        q.append((head,0))
        maxT=0
        while q:
            for i in range(len(q)):
                emp,t = q.popleft()
                maxT=max(maxT, t)
                for nei in graph[emp]:
                    neiT=informTime[emp]+t
                    q.append((nei,neiT))
        return maxT
